tests/metawolf/utils.py

```diff
@@ -0,0 +1,179 @@
+    """That whether or not a string can be represented as a boolean."""
+    for s in ['YeS', 'tRue', 'T', 'y']:
+      self.assertTrue(utils.MetawolfUtils().Str2Bool(s))
+    for s in ['nO', 'fAlsE', 'f', 'N']:
+      self.assertFalse(utils.MetawolfUtils().Str2Bool(s))
+    self.assertIsNone(utils.MetawolfUtils().Str2Bool('foo'))
+
+  def testGetType(self) -> None:
+    """Test that the type of a string is inferred correctly."""
+    self.assertEqual(bool, utils.MetawolfUtils().GetType('y'))
+    self.assertEqual(int, utils.MetawolfUtils().GetType('0'))
+    self.assertEqual(float, utils.MetawolfUtils().GetType('1.'))
+
+  def testCastToType(self) -> None:
+    """Test that a string can be cast to the desired type."""
+    self.assertTrue(utils.MetawolfUtils().CastToType('y', bool))
+    self.assertEqual(0, utils.MetawolfUtils().CastToType('0', int))
+    self.assertEqual(1., utils.MetawolfUtils().CastToType('1.', float))
+
+  def testMarshal(self) -> None:
+    """Test that a session settable object marshals correctly."""
+    self.assertEqual(MOCK_SESSION_SETTABLE_JSON,
+                     utils.MetawolfUtils().Marshal(MOCK_SESSION_SETTABLE))
+
+  def testUnmarshal(self) -> None:
+    """Test that a JSON dict of a session settable unmarshalls correctly."""
+    unmarshalled = utils.MetawolfUtils().Unmarshal(MOCK_SESSION_SETTABLE_JSON)
+    self.assertEqual(
+        MOCK_SESSION_SETTABLE.session_id,
+        unmarshalled.session_id)
+    self.assertEqual(
+        MOCK_SESSION_SETTABLE.description,
+        unmarshalled.description)
+    self.assertEqual(
+        MOCK_SESSION_SETTABLE.GetValue(),
+        unmarshalled.GetValue())
+
+  @mock.patch('json.loads')
+  @mock.patch(
+      'builtins.open', new_callable=mock_open, read_data=str(MOCK_SESSION_JSON))
+  @mock.patch('os.path.exists')
+  @typing.no_type_check
+  def testReadSessionFromFile(self, mock_exists, m_open, mock_json) -> None:
+    """Test that the session file is read correctly."""
+    mock_exists.return_value = True
+    mock_json.return_value = MOCK_SESSION_JSON
+    s = utils.MetawolfUtils().ReadSessionFromFile()
+    self.assertIn('session_test', s)
+    json_session_settable = s['session_test']['recipe_test'][
+      'session_test-recipe_test-param_name']
+    # Check a few fields for equality
+    self.assertEqual(MOCK_SESSION_SETTABLE.session_id,
+                     json_session_settable.session_id)
+    self.assertEqual(MOCK_SESSION_SETTABLE.description,
+                     json_session_settable.description)
+    self.assertEqual(MOCK_SESSION_SETTABLE.GetValue(),
+                     json_session_settable.GetValue())
+
+    s = utils.MetawolfUtils().ReadSessionFromFile(unmarshal=False)
+    self.assertIn('session_test', s)
+    self.assertEqual(
+        MOCK_SESSION_SETTABLE_JSON,
+        s['session_test']['recipe_test']['session_test-recipe_test-param_name'])
+
+  @mock.patch('dftimewolf.lib.recipes.manager.RecipesManager.Recipes')
```

Comment:
Same comment as above - it would be better to use the actual Recipes object (even this means providing a path to test recipes; there's already some you can use / extended in `tests/test_modules/test_recipe.py`)

Reply:
The mocked return value (`MOCK_DFTIMEWOLF_RECIPE`)'s underlying type is an actual Recipe object; it's just a dummy one (this was to simplify the built command). I'm not sure I understand what you want different here?

Reply:
As discussed, changed